test/adapters/http_test.exs

```diff
@@ -1,14 +1,33 @@
 defmodule Shopify.Adapters.HTTPTest do
   use ExUnit.Case, async: true
 
-  alias Shopify.{Adapters.HTTP, Error, Product}
+  alias Shopify.{Adapters.HTTP, Session, Request, Error, Product}
 
   test "it handles httpoison errors" do
     result = {:error, %HTTPoison.Error{id: nil, reason: :econnrefused}}
 
     assert {:error, %Error{} = error} = HTTP.handle_response(result, %Product{})
-
     assert error.source == :httpoison
     assert error.reason == :econnrefused
   end
+
+  test "it will use session req_opts" do
+    request =
+      Session.new()
+      |> Session.put_req_opt(:recv_timeout, 0)
+      |> Request.new("/", %{}, Product)
+
+    assert {:error, %Error{} = error} = HTTP.get(request)
+    assert error.source == :httpoison
+    assert error.reason == :timeout
+
```

Comment:
I think this test absolutely does the job, but it raises a few questions.

The test is very tightly coupled to the implementation, i wonder if it would make sense to mock httpoison here and just test if it receives the opts.

The test title says it should use the opts, but it is actually testing if the opts are passed to httpoison. Should we change the test title here?